backend/notes.go

```diff
@@ -0,0 +1,303 @@
+		// from the ETH account), and the user cannot modify the ERC20 token account name.
+		ethCoin, isETHCoin := account.Coin().(*eth.Coin)
+		isERC20 := isETHCoin && ethCoin.ERC20Token() != nil
+		if accountName != "" && !isERC20 {
+			for _, signingConfig := range account.Config().Config.SigningConfigurations {
+				entry := bip329Entry{
+					Type:  bip329TypeXpub,
+					Ref:   signingConfig.ExtendedPublicKey().String(),
+					Label: accountName,
+					// We don't technically need this for the import, as we assume no two accounts
+					// in the BitBoxApp contain the same xpub. Note that ERC20-token accounts have
+					// the same xpub as the parent ETH account, but they are auto-derived and do not
+					// exist in the persisted config where the account name is stored.
+					//
+					// We add it anyway just in case, as it can't hurt and is more explicit.
+					BitBoxApp: &bip329BitBoxApp{
+						CoinCode:    account.Config().Config.CoinCode,
+						AccountCode: account.Config().Config.Code,
+					},
+				}
+				if err := json.NewEncoder(writer).Encode(entry); err != nil {
+					return err
+				}
+			}
+		}
+
+		notesData := account.Notes().Data()
+		for txID, note := range notesData.TransactionNotes {
+			entry := bip329Entry{
+				Type:  bip329TypeTx,
+				Ref:   txID,
+				Label: util.TruncateString(note, notes.MaxNoteLen),
+				// We use this to aid the import back into the BitBoxApp, as the same txID can be in
+				// multiple accounts with different labels (e.g. and ERC20 token in the ERC20 token
+				// account as well as in the parent ETH account), and the origin label is not a good
+				// fit (see docstring of `bip329BitBoxApp`).
+				BitBoxApp: &bip329BitBoxApp{
+					CoinCode:    account.Config().Config.CoinCode,
+					AccountCode: accountCode,
+				},
+			}
+			if err := json.NewEncoder(writer).Encode(entry); err != nil {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// ExportNotes exports the transaction and account labels of all accounts of all
+// connected/remembered keystores. Deactivated accounts are included in the export, except for
+// deactivated ERC-20 accounts. We export to a file using an extended version of BIP-329:
+// https://github.com/bitcoin/bips/blob/master/bip-0329.mediawiki
+func (backend *Backend) ExportNotes() error {
+	exportsDir, err := utilcfg.ExportsDir()
+	if err != nil {
+		return err
+	}
+	name := fmt.Sprintf("%s-notes.txt", time.Now().Format("2006-01-02-at-15-04-05"))
+	suggestedPath := filepath.Join(exportsDir, name)
+	path := backend.Environment().GetSaveFilename(suggestedPath)
+	if path == "" {
+		return errp.ErrUserAbort
+	}
+	err = func() error {
+		file, err := os.Create(path)
+		if err != nil {
+			return err
+		}
+		defer func() { _ = file.Close() }()
+
+		writer := bufio.NewWriter(file)
+		if err := backend.exportNotes(writer); err != nil {
+			return err
+		}
+		return writer.Flush()
+	}()
+	if err != nil {
+		return err
+	}
+
+	if runtime.GOOS == "android" || runtime.GOOS == "ios" {
```

Comment:
Because this invokes the 'Share' workflow on Android (and I presumed it would work similarly on ios), while on the computer, there is no need to open the file, it is saved to the location the user chose.